docker/ci/Dockerfile

```diff
@@ -1,27 +1,87 @@
-FROM golang:1.8-alpine
+FROM centos:6
 
-RUN apk --no-cache add bash clang git g++ make nodejs openjdk8 python perl \
-    ruby ruby-bigdecimal ruby-bundler ruby-io-console ruby-irb ruby-json openssl ca-certificates \
-    && echo "http://dl-cdn.alpinelinux.org/alpine/v3.5/main" >> /etc/apk/repositories \
-    && apk --no-cache add "protobuf>3.1.0" \
-    && go get -u github.com/golang/protobuf/proto \
-    && go get -u github.com/golang/protobuf/protoc-gen-go \
-    && go get -u google.golang.org/grpc \
-# TODO: remove third party glibc (as well as openssl and ca-certificates) package when protobuf works with musl (next release)
-    && wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://raw.githubusercontent.com/sgerrand/alpine-pkg-glibc/master/sgerrand.rsa.pub \
-    && wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.23-r3/glibc-2.23-r3.apk \
-    && apk add glibc-2.23-r3.apk \
-    && MAVEN_VERSION=3.3.9 \
+ENV CHAIN="/go/src/chain" \
+    BUNDLE_APP_CONFIG="/usr/local/bundle" \
+    GEM_HOME="/usr/local/bundle" \
+    GOLANG_VERSION="1.8" \
+    GOLANG_URL="https://golang.org/dl/go1.8.linux-amd64.tar.gz" \
+    GOLANG_SHA256="53ab94104ee3923e228a2cb2116e5e462ad3ebaeea06ff04463479d7f12d27ca" \
+    GOPATH="/go" \
+    GOROOT="/usr/local/go" \
+    JAVA_HOME="/opt/jdk1.7.0_79" \
+    JRE_HOME="/opt/jdk1.7.0_79/jre" \
+    MAVEN_HOME="/usr/share/maven" \
+    MAVEN_VERSION="3.3.9" \
+    PATH="$PATH:/usr/local/bundle/bin:/go/bin:/usr/local/go/bin:/opt/jdk1.7.0_79/bin:/opt/jdk1.7.0_79/jre/bin" \
+    RUBY_MAJOR="2.0" \
+    RUBY_VERSION="2.0.0-p594"
```

Comment:
Minor but I'd slightly prefer leaving a trailing backslash on the last line here. That way it's less likely that additional lines will forget it, and it makes the diff less noisy.